packages/blocks/src/root-block/config/editor-settings.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  BrushSchema,
+  ConnectorSchema,
+  EdgelessTextSchema,
+  NoteSchema,
+  ShapeSchema,
+} from '@blocksuite/affine-shared/utils';
+import { z } from 'zod';
+
+export const EditorSettingSchema = z.object({
+  connector: ConnectorSchema,
+  brush: BrushSchema,
+  shape: ShapeSchema,
+  'affine:edgeless-text': EdgelessTextSchema,
+  'affine:note': NoteSchema,
+});
+
+export type EditorSetting = z.infer<typeof EditorSettingSchema>;
```

Comment:
I think this should be in `affine-shared` package.

Reply:
Here `EditorSetting` is only used by `root-config`. I think we shouldn't provide all fields of `lastProps`t o the user as configuration. Currently they are not equal, and consist of different kinds of schemas.